GVRf/Framework/src/org/gearvrf/GVRResourceVolume.java

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import android.os.Environment;
+
+/**
+ * Models a file system which supports stream I/O.
+ */
+public class GVRResourceVolume {
+    private static final String TAG = GVRResourceVolume.class.getSimpleName();
+
+    public enum VolumeType {
+        ANDROID_ASSETS,
+        ANDROID_SDCARD,
+        LINUX_FILESYSTEM,
+        NETWORK,
+    }
+
+    protected GVRContext gvrContext;
+    protected VolumeType volumeType;
+    protected String defaultPath;
+
+    public GVRResourceVolume(GVRContext gvrContext, VolumeType volume) {
+        this(gvrContext, volume, null);
+    }
+
+    public GVRResourceVolume(GVRContext gvrContext, VolumeType volumeType, String defaultPath) {
+        this.gvrContext = gvrContext;
+        this.volumeType = volumeType;
+        this.defaultPath = defaultPath;
+    }
+
+    /**
+     * Opens a file from the volume. The filePath is relative to the defaultPath.
+     *
+     * @param filePath          File path of the resource to open.
+     *
+     * @throws IOException
+     */
+    public GVRAndroidResource openResource(String filePath) throws IOException {
```

Comment:
(1)
For the sake of the discussion, the various cases could have been different static factory methods in GVRAndroidResource:
```
GVRAndroidResource r = GVRAndroidResource.assetsResource(..)
GVRAndroidResource r = GVRAndroidResource.filesystemResource(..)
GVRAndroidResource r = GVRAndroidResource.externalStorageResource(..)
GVRAndroidResource r = GVRAndroidResource.networkResource(..)
```

People are already having to deal with loadTexture and loadMesh so they ideally should have to be familiar with one way of representing a resource (given the fact that we went through the trouble to come up with GVRAndroidResource).